packages/flutter_tools/lib/src/runner/local_engine.dart

```diff
@@ -176,20 +187,25 @@ class LocalEngineLocator {
     return 'host_$tmpBasename';
   }
 
-  EngineBuildPaths _findEngineBuildPath(String? localEngine, String? localWebSdk, String enginePath) {
+  EngineBuildPaths _findEngineBuildPath({
+    required String engineSourcePath,
+    String? localEngine,
+    String? localWebSdk,
+    String? localHostEngine,
+  }) {
     if (localEngine == null && localWebSdk == null) {
       throwToolExit(_userMessages.runnerLocalEngineOrWebSdkRequired, exitCode: 2);
     }
 
     String? engineBuildPath;
     String? engineHostBuildPath;
     if (localEngine != null) {
-      engineBuildPath = _fileSystem.path.normalize(_fileSystem.path.join(enginePath, 'out', localEngine));
+      engineBuildPath = _fileSystem.path.normalize(_fileSystem.path.join(engineSourcePath, 'out', localEngine));
       if (!_fileSystem.isDirectorySync(engineBuildPath)) {
         throwToolExit(_userMessages.runnerNoEngineBuild(engineBuildPath), exitCode: 2);
       }
 
-      final String basename = _fileSystem.path.basename(engineBuildPath);
+      final String basename = localHostEngine ?? _fileSystem.path.basename(engineBuildPath);
```

Comment:
I think zach suggested requiring the new flag. Could we at this point check that `localHostEngine` is not null, and `throwToolExit('The --local-host-engine CLI option is required when using --local-engine')`?

Reply:
While I think this is the right direction (i.e. we can rip out trying to derive the host engine), it would be a change to _all_ workflows of folks that contribute to the engine, or use local engine builds for debugging framework issues. It could also impact CI.

If we're not worried about any of that, happy to do so. Otherwise we might want to at least do a few week "bake" period (perhaps we could add a warning message that `--local-engine` will soon require `--local-host-engine`?).

Reply:
Ahh you know what, good point, I think this *would* break certain CI tests. Adding a warning SGTM.

cc @whesse, do you use the --local-engine option?

Reply:
Filed https://github.com/flutter/flutter/issues/132245 and I'll also add a warning message in the tool in this PR.

Reply:
The monorepo builds do not, but the legacy triple-head building on Golem and perhaps some people's local workflows may use it.

Actually, in our internal system, I see something which could be fixed by this:
```
// Work around bug where --local-engine=android_$mode_arm64 makes it try to
--
  | // use host_$mode_arm64, which is wrong.
  | if (machineType == const MachineType(OS.android, Arch.armv8)) {
  | build.addCustomCommand("ln -s host_$mode out/host_${mode}_arm64");
  | }
```

Reply:
Nice! Thanks for sharing, we should definitely be able to fix this.

Reply:
But our other uses of --local-engine will need to add the proper --local-host-engine, before it is made mandatory.  It would be less breaking if it is optional first, and made mandatory later.